### DISCUSSION TITLE
How to align my tempo and the app tempo ?

### DISCUSSION THREAD

**Discussion by YetAnotherMe** [ryohey/signal#289] [Q&A]
Hi, 

I play half notes on an external keyboard, save as midi and open in this app. They don’t occupy half a bar each, but less. Makes sense : how could the app know what’s a bar for me ?
So I’d like to tell the app : look this duration (ex two half notes) = a bar. So that everything is well aligned and makes sense musically. 

i tried playing with the tempo setting but it didn’t help. Anything else I could do ?

note: midi file created with midi rec app on iPhone.

**Comment by robertnhart** [comment]
Unfortunately, it looks like Signal can not do this.

...

In MIDI files, the time position and duration of notes have both a **musical time** in quarter notes and an **absolute time** in seconds.

The tempo (in quarter notes per minute) is a conversion factor between the musical time in quarter notes and the absolute time in seconds.

In most MIDI file software, when you change the tempo, the notes will keep the same musical time in quarter notes and their absolute time in seconds will be changed according to the tempo conversion factor.

In your case, it sounds like you want to change the notes' musical time in quarter notes and keep the absolute time in seconds the same. Unfortunately, Signal can not do this.

...

If you have a Windows computer, check out the following post I wrote for some ideas about how to do this kind of change using the MIDI file software [Sekaiju](https://openmidiproject.opal.ne.jp/Sekaiju_en.html) or [Cakewalk](https://www.bandlab.com/products/cakewalk).

Go to [Re: Change tempo of mid file without changing perceived timing](https://midi.org/community/midi-specifications/change-tempo-of-mid-file-without-changing-perceived-timing#post-5676) (midi.org). Scroll down to the paragraph just before "Idea 1" that starts with "Here are two ideas how to change the tempo but keep the notes at the same positions in seconds." Read that intro paragraph and the sections "Idea 1" and "Idea 2".

**Reply by YetAnotherMe** [reply]
That makes perfectly sense. Thank you very much for your detailed answer.
Best,
Eric
